src/scraper.rs

```diff
@@ -313,19 +331,58 @@ impl Scraper {
         std::thread::sleep(delay_duration);
     }
 
-    /// If a URL should be visited, or does it belong to another domain
-    fn should_visit(url: &str, base: &Url) -> bool {
+    /// If a URL should be visited (ignores `mail:`, `javascript:` and other pseudo-links)
+    fn should_visit(url: &str) -> bool {
         match Url::parse(url) {
+            /* The given candidate is a valid URL, and not a relative path to
+             * the next one. Therefore, we have to check if this URL is valid.
+             * If it is, we should visit it.
+             */
+            Ok(not_ok) => not_ok.has_host() && !not_ok.cannot_be_a_base(),
+
+            /* Since we couldn't parse this "URL", then it must be a relative
+             * path or a malformed URL. If the URL is malformed, then it will
+             * be handled during the join() call in run() */
+            Err(_) => true,
+        }
+    }
+
+    /// Replaces `///` with `//`
+    /// And `//` with `https://`
+    /// Without this function, if url is `///tools.wmflabs.org/`, the app crashes.
+    fn normalize_url(data: String) -> String {
+        let mut temp = data.clone();
+
+        if temp.starts_with("///") {
+            temp = data.replacen("///", "https://", 1);
+        }
+        if temp.starts_with("//") {
+            temp = data.replacen("//", "https://", 1);
+        }
+        temp
+    }
```

Comment:
```suggestion
    fn normalize_url(data: String) -> String {
        if data.starts_with("///") {
            data.replacen("///", "https://", 1);
        } else if data.starts_with("//") {
            data.replacen("//", "https://", 1);
        } else {
           data
        }
    }
```
Is more idiomatic I think. This might not compile, check before applying

Reply:
`data.replacen()` does not change the `data` variable as far as I understand. Also, `else if` should now be there, as there can be both `///` and `//` in the link.

Also this won't compile since only the last `else` clause returns some value

Reply:
I have just committed this code:

```rust
    fn normalize_url(url: String) -> String {
        if url.starts_with("///") {
            return url.replacen("///", "https://", 1);
        } else if url.starts_with("//") {
            return url.replacen("//", "https://", 1);
        }
        url
    }
```